test/remote.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	remoteimg "github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// CreateImage will push a new OCI image artifact with the provided raw data object as a layer and return the full image
+// reference with a digest to fetch the image. Key must be specified as [lowercase kind]/[object name]. The image ref
+// with a digest is returned.
+func CreateImage(ref string, objs ...runtime.Object) (string, error) {
```

Comment:
https://github.com/tektoncd/experimental/tree/master/oci should be update accordingly I think.

Reply:
The goal is to abandon the experimental tool so we can go in and remove it.

Reply:
The `experimental` tool will still have value for a while, to be able to test this feature we will need to be able to create such images, having the `oci` tool in experimental is the best way to achieve that while we work on this.

To re-iterate a little bit what `tektoncd/experimental` is for : quicker *dev to in `master`* cycle. It's way easier to iterate quickly in experimental, demo it and later on integrating it in the other projects (here, on cli, …)

Reply:
Discussed offline. Our hope is to get in a "spec" document in the next PR to act as the stopgap so users will be able to create their own solution while waiting for the official Tekton solution.